types/msg.go

```diff
@@ -77,8 +79,41 @@ type BankMsg struct {
 // SendMsg contains instructions for a Cosmos-SDK/SendMsg
 // It has a fixed interface here and should be converted into the proper SDK format before dispatching
 type SendMsg struct {
-	ToAddress   string `json:"to_address"`
-	Amount      Coins  `json:"amount"`
+	ToAddress string `json:"to_address"`
+	Amount    Coins  `json:"amount"`
+}
+
+type IbcMsg struct {
+	Ics20Transfer *Ics20TransferMsg `json:"ics20_transfer,omitempty"`
+	SendPacket    *SendPacketMsg    `json:"send_packet,omitempty"`
+	CloseChannel  *CloseChannelMsg  `json:"close_channel,omitempty"`
+}
+
+type Ics20TransferMsg struct {
+	ChannelID string `json:"channel_id"`
+	ToAddress string `json:"to_address"`
+	Amount    Coins  `json:"amount"`
+}
```

Comment:
Two fields are missing. I don't think we can come up with reasonable defaults as they are on the application level:
* `TimeoutHeight    *IbcTimeoutHeight` // a block timeout height to prevent the tx from being committed past a certain height
* `TimeoutTimestamp uint64` //  absolute timeouts.

Reply:
I thought these were per-channel level now, not per-packet. (can you double check latest state of IBC?)
In any case, yes, the transfer msg and the send packet msg need the same info for timeout.

Reply:
This comment is a bit confusing: https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/proto/ibc/applications/transfer/v1/tx.proto#L34

But in general, it seems like both of these *may* be set. 0, 1, or 2. And that we need a clear way to mark zero/unset

Reply:
From the sdk code I can see:
* height is empty when  `h.RevisionNumber == 0 && h.RevisionHeight == 0`
* timestamp is empty when `==0`
One of them must be set. I assume that you _can_ set both.